model/task/task.go

```diff
@@ -2797,6 +2757,38 @@ func (t *Task) SetResetWhenFinished() error {
 	)
 }
 
+// SetResetWhenFinishedWithInc requests that a task (that was marked to
+// automatically reset when finished via the agent status server) reset itself
+// when finished. It will also increment the number of automatic resets the task
+// has performed.
+func (t *Task) SetResetWhenFinishedWithInc() error {
+	if t.ResetWhenFinished {
+		return nil
+	}
+	err := UpdateOne(
+		bson.M{
+			IdKey:                 t.Id,
+			AbortedKey:            bson.M{"$ne": true},
+			IsAutomaticRestartKey: bson.M{"$ne": true},
+		},
+		bson.M{
+			"$set": bson.M{
+				ResetWhenFinishedKey:  true,
+				IsAutomaticRestartKey: true,
+			},
+			"$inc": bson.M{
+				NumAutomaticRestartsKey: 1,
+			},
+		},
+	)
+	if err != nil {
```

Comment:
This is going to return an error for aborted tasks, since its not going to match anything. Is that okay?

Reply:
I think we should return an error there to inform in the logs [here](https://gist.github.com/hadjri/aed360cffb5610ec81dd256cae5c354a#file-command-go-L221) that the auto marking for reset did not go through. That being said I updated this function with a more descriptive error if the task has been aborted, rather than propagating a vague "matched no documents" error down to the user